scripts/generate-idls.mjs

```diff
@@ -15,6 +15,16 @@ getProgramFolders().forEach((folder) => {
     programName: cargo.package.name.replace(/-/g, '_'),
     programId: cargo.package.metadata.solana['program-id'],
     idlDir: programDir,
+    idlHook: (idl) => ({
+      ...idl,
+      instructions: idl.instructions.map((instruction) => ({
+        ...instruction,
+        discriminant: {
+          ...instruction.discriminant,
+          type: "u32", // The legacy native program only accepts 4-byte instruction discriminants.
```

Comment:
Paging @lorisleiva, Shank wizard. Is there a way to convince Shank to produce 4-byte instruction discriminators other than to hot-patch the idl?

Reply:
You can maybe update the comment here to say "this program" instead of "the legacy native program", since they are both the same, but idc.

Reply:
Sadly no, Shank doesn't support custom enum discriminants. You could use a Codama visitor to update this but since we're currently saving the Anchor IDL and not the Codama one, your solution is currently the best thing to do.